controllers/routemonitor/routemonitor_supplement.go

```diff
@@ -235,7 +236,24 @@ func (r *RouteMonitorReconciler) EnsureRouteURLExists(route routev1.Route, route
 	}
 
 	if currentRouteURL != "" && extractedRouteURL != currentRouteURL {
-		r.Log.V(3).Info("RouteURL mismatch: currentRouteURL and extractedRouteURL are not equal, taking extractedRouteURL as source of truth")
+		client := &http.Client{
+			CheckRedirect: func(req *http.Request, via []*http.Request) error {
+				return http.ErrUseLastResponse
+			},
+		}
+
+		resp, err := client.Head(extractedRouteURL)
+		if err != nil {
+			r.Log.V(3).Info("Failed to make HEAD request to the URL")
```

Comment:
Looks the requests fails only logs an `info` message, it would be better we can improve the error handling for this HEAD request to be more visible for debugging purposes. WDYT?